### DISCUSSION TITLE
Global mutate doesn't work

### DISCUSSION THREAD

**Discussion by nimabrownlee** [vercel/swr#746] [Q&A]
# Bug report

## Description / Observed Behavior

I want to revalidate the data/send a new request using the global mutate function.

## Expected Behavior

To send a new request.

## Repro Steps / Code Example

I'll just post a code snippet as I am not doing anything fancy. 

```
  const { data, error } = useSWR(
    '/api/tasks' +
      encodeQueryString({
        limit: 3,
        contactId: id,
      }),
  ) 
// /api/tasks?limit=3&contactId=30404
```

and then when I use the global mutate function

`mutate('/api/tasks')`

Now I am not adding the query and I assume thats how it should work. Because I don't have access to contactId in every instance.

**Comment by promer94** [comment]
if you have two hooks mounted
```js
const { data, error } = useSWR('/api/tasks')
const { data, error } = useSWR('/api/tasks?limit=3&contactId=30404')
```
call ```mutate('/api/tasks')``` will only revalidate the first hooks

For more informations, you could check the [mutation section](https://swr.vercel.app/docs/mutation) in [docs](https://swr.vercel.app)

**Comment by nimabrownlee** [comment]
@promer94 Then isn't the global mutate pretty useless? Obviously I can't make `contactId` accessible everywhere. I think there should be a change in the implementation so that it ignores queries.

**Comment by shuding** [comment]
This is expected exactly like @promer94 said, so you won't accidentally request something unintended.

However if you want to revalidate some data without knowing the key, you can only use an internal API for now (might be changed in the future): https://github.com/vercel/swr/blob/master/src/cache.ts. You can basically get all the keys from there and find the ones with the same API path and mutate them.

We can work on a new API for this case (like accepting a filter function for `mutate`), feel free to open a feature request 👍
